generative_data_prep/utils/logger.py

```diff
@@ -0,0 +1,116 @@
+            breakpoint()
+            cls.instance = super(Logger, cls).__new__(cls)
+            cls.instance._logger = logging.getLogger("generative_data_prep_logger")
+            cls.instance._logger.setLevel(logging.DEBUG)
+            console_handler = logging.StreamHandler()  # This is your console handler
+            console_handler.setLevel(logging.INFO)
+            formatter = logging.Formatter("%(message)s")
+            console_handler.setFormatter(formatter)
+            cls.instance._logger.addHandler(console_handler)
+            return cls.instance
+        return cls.instance
+
+    @classmethod
+    def info(cls, message, print_sep_str=True):
+        """Log at the info level."""
+        if print_sep_str:
+            cls._logger(SEP_STR)
+        cls._logger.info(message)
+
+    @classmethod
+    def error(cls, message, print_sep_str=True):
+        """Log at the error level."""
+        if print_sep_str:
+            cls._logger(SEP_STR)
+        cls._logger.info(message)
+
+    @classmethod
+    def warning(cls, message, print_sep_str=True):
+        """Log at the warning level."""
+        if print_sep_str:
+            cls._logger(SEP_STR)
+        cls._logger.info(message)
+
+    @classmethod
+    def debug(cls, message, print_sep_str=True):
+        """Log at the debug level."""
+        if print_sep_str:
+            cls._logger(SEP_STR)
+        cls._logger.info(message)
+
+    @classmethod
+    def log_current_datetime(cls):
+        """Log at the current date and time."""
+        current_datetime = datetime.datetime.now()
+        cls._logger.debug(f"Time of execution: {current_datetime}")
+
+    @classmethod
+    def log_git_commit_hash(cls):
+        """Log the current git commit hash."""
+        repo = git.Repo(search_parent_directories=True)
+        sha = repo.head.object.hexsha
+        cls._logger.debug("Running Generative Data Prep repository: https://github.com/sambanova/generative_data_prep/")
+        cls._logger.debug(f"Git commit hash: {sha}")
+
+    @classmethod
+    def log_input_args(cls, args):
+        """Log the input arguments."""
+        cls._logger.debug(SEP_STR)
+        cls._logger.debug("Logging command line input flags.")
+        argument_dict = vars(args)
+        for arg, value in argument_dict.items():
+            cls._logger.debug(f"{arg}: {value}")
+
+    @classmethod
+    def add_handler(cls, handler):
+        """Add a handler to the logger."""
+        cls._logger.addHandler(handler)
+
+    @classmethod
+    def log_metrics(cls, metrics):
+        """Log the metrics table."""
+        cls._logger.info(f"{cls._header('Metrics')}\n{metrics}\n{cls._header('Complete')}")
+
+    @classmethod
+    def _header(cls, header_name: str):
+        """Create a header out of the header_name string."""
+        half_sep_str = int((len(SEP_STR) - len(header_name)) / 2) * "-"
+        return half_sep_str + header_name + half_sep_str
+
+
+logger = Logger()
```

Comment:
please delete

Reply:
sorry take this back, your way is better

Reply:
actually, I'm not sure, let's discuss tomorrow and weigh the pros and cons of each.  If we do it this way we don't need a Singleton class.